libr/core/canal.c

```diff
@@ -2753,39 +2753,76 @@ R_API void r_core_recover_vars(RCore *core, RAnalFunction *fcn, bool argonly) {
 	return;
 }
 
-R_API RList* r_core_anal_graph_to(RCore *core, ut64 addr, int n) {
-	RAnalBlock *bb, *root = NULL, *dest = NULL;
-	RListIter *iter, *iter2;
-	RList *list2 = NULL, *list = NULL;
-	RAnalFunction *fcn;
+static bool anal_path_exists(RCore *core, ut64 from, ut64 to, RList *bbs, int depth, RList *state) {
+	RAnalBlock *bb = r_anal_bb_from_offset (core->anal, from);
+	RListIter *iter = NULL;
+	RList *refs = NULL;
+	RAnalRef *refi;
+	RAnalFunction *fcn = NULL;
+	RAnalFunction *cur_fcn = NULL;
+
+	if (depth < 1) {
+		eprintf ("going too deep\n");
+		return false;
+	}
 
+	if (!bb || !bbs) {
```

Comment:
r_return_val_if_fail?

Reply:
r_return_val_if_fail only for `bbs`. `bb` needs to be checked with `if` as you've already done, because it is a runtime error that needs to be checked.

Reply:
I can't here, because I get a lot of warnings with this. I just want to silently ignore non predictable branches.

Reply:
oh sorry, I did not see your comment @ret2libc, I can do that !

Reply:
done